packages/lab/src/list-next/useList.ts

```diff
@@ -0,0 +1,248 @@
+  const select = useCallback(
+    (event: SyntheticEvent<HTMLLIElement>) => {
+      const newValue = event.currentTarget.dataset.value;
+      if (event.type === "click") {
+        setShowFocusRing(false);
+      }
+      const activeOptions = getOptions();
+      const isActiveOption =
+        activeOptions.findIndex((i) => i.id === event.currentTarget.id) !== -1;
+      if (newValue && selectedItem !== newValue && isActiveOption) {
+        selectItem(event.currentTarget);
+        onChange?.(event, { value: newValue });
+      }
+    },
+    [selectItem, selectedItem, onChange, getOptions]
+  );
+
+  const isSelected = useCallback(
+    (id: string) => selectedItem === id,
+    [selectedItem]
+  );
+
+  const isFocused = useCallback(
+    (id: string | undefined) => activeDescendant === id && showFocusRing,
+    [activeDescendant, showFocusRing]
+  );
+
+  const getActiveItem = () => {
+    const activeOptions = getOptions();
+    const activeIndex = activeOptions.findIndex(
+      (i) => i.id === activeDescendant
+    );
+    return activeOptions[activeIndex !== -1 ? activeIndex : 0];
+  };
+
+  // HANDLERS
+  const blurHandler = () => {
+    setShowFocusRing(false);
+  };
+
+  const mouseDownHandler = () => {
+    // When list gets focused, we can't guarantee that focus happens after click event.
+    // If first focus (where !activeDescendant) happens from a click, list shouldn't render focus ring in the first element.
+    setShowFocusRing(false);
+  };
+
+  // takes care of focus when using keyboard navigation
+  const focusHandler = () => {
+    const activeElement = getActiveItem();
+    focusAndMoveActive(activeElement);
+  };
+
+  // takes care of keydown when using keyboard navigation
+  const keyDownHandler = (event: KeyboardEvent<HTMLUListElement>) => {
+    const { key } = event;
+    const currentItem = getActiveItem();
+    let nextItem = currentItem;
+    if (!currentItem) {
+      return;
+    }
+    switch (key) {
+      case "ArrowUp":
+      case "ArrowDown":
+        nextItem =
+          key === "ArrowUp"
+            ? findPreviousOption(currentItem, 1)
+            : findNextOption(currentItem, 1);
+
+        if (nextItem && nextItem !== currentItem) {
+          event.preventDefault();
+          focusAndMoveActive(nextItem);
+        }
+        break;
+      case "Home":
+        event.preventDefault();
+        focusFirstItem();
+        break;
+      case "End":
+        event.preventDefault();
+        focusLastItem();
+        break;
+      case " ":
+      case "Enter":
+        event.preventDefault();
+        nextItem && selectItem(nextItem);
+        break;
+      default:
+        break;
+    }
+  };
+
+  // CONTEXT
+  const contextValue = useMemo(
+    () => ({
+      disabled,
+      id,
+      select,
+      isSelected,
+      isFocused,
+    }),
+    [disabled, id, select, isSelected, isFocused]
+  );
+
+  return {
+    focusHandler,
+    keyDownHandler,
+    blurHandler,
+    mouseDownHandler,
```

Comment:
Any reason we don't want to name these onKeyDown, onBlur... etc?

Reply:
We should probably discuss a standard, it could be something like:
on* for props or parameters and handle* for the functions